src/platform_impl/macos/view.rs

```diff
@@ -98,6 +125,32 @@ pub unsafe fn set_ime_position(ns_view: id, input_context: id, x: f64, y: f64) {
     let _: () = msg_send![input_context, invalidateCharacterCoordinates];
 }
 
+pub unsafe fn set_ime_allowed(ns_view: id, ime_allowed: bool) {
+    let state_ptr: *mut c_void = *(*ns_view).get_mut_ivar("winitState");
+    let state = &mut *(state_ptr as *mut ViewState);
+    if state.app_allowed_ime == ime_allowed {
+        return;
+    }
+    state.app_allowed_ime = ime_allowed;
+    if !ime_allowed {
```

Comment:
Could you early return here? Also, I think it's more logical to test `state.app_allowed_ime` after you've assigned to it, but it's a matter of preference.

Reply:
I do return early. It's 3 lines above this one, at line 132.

Reply:
I mean that can do for this line as well afaics, since the rest of the function is inside this if.

Reply:
I don't think it's an improvement but I also don't think it's worth the time arguing about it, so I changed it.